### DISCUSSION TITLE
index.html template: what renders the initial script @vite-plugin-checker-runtime @vite/client ?

### DISCUSSION THREAD

**Discussion by space88man** [quasarframework/quasar#17491] [General - Components / Directives / etc] [answered]
I am coming from a quasar-less vite/vue3 workflow: in the template what plugin is responsible for 
rendering this:
```
    <script type="module">
import { inject } from "/@vite-plugin-checker-runtime";
inject({
  overlayConfig: {},
  base: "/",
});
</script>
    <script type="module" src="/@vite/client"></script>
```

I think I  understand that `<!-- quasar:entry-point -->` is a marker used to render `<div id="q-app">...`.

Now I'm trying to grok what triggers the first script element: grep'ing `html-template.js` did not enlighten re `@vite-plugin-checker-runtime` and `@vite/client` - although I did find some `injectXXX` methods for SSR and dev tools.

**Comment by yusufkandemir** [comment, marked as answer]
https://github.com/fi3ework/vite-plugin-checker is injecting the first part. It's installed if you choose linter/checker during project creation.

Vite itself is injecting `@vite/client`.

Both injections happen at the Vite level after it starts processing the `index.html`
